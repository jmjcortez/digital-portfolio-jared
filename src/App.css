.App {
  display: flex;
}

.menu {
  display: none;
}

@media (max-width: 800px) {
  .menu {
    display: block;
    position: absolute;
    top: 30px;
    left: 30px;
    font-size: 30px;
    z-index: 11;
  }  
}
