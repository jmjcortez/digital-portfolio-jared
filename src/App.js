import { useState, useEffect } from 'react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';

import './App.css';

import About from './components/about/About';
import Contact from './components/contact/Contact';
import Illustrations from './components/illustrations/Illustrations';
import Sidebar from './components/sidebar/Sidebar';
import { BiMenu } from 'react-icons/bi';

function App() {

  const [width, setWidth] = useState(window.innerWidth);

  const handleWindowSizeChange = () => {
    setWidth(window.innerWidth);
  }

  useEffect(() => {
    window.addEventListener('resize', handleWindowSizeChange);
    return () => {
      window.removeEventListener('resize', handleWindowSizeChange);
    }
  }, [])
  
  const isDesktop = width > 844;

  const [showSideBar, setshowSideBar] = useState(isDesktop);

  const toggleSideBar = () => {
    setshowSideBar(!showSideBar)
  }

  return (
    <div className="App" data-testid='App'>
      <BrowserRouter>
        <BiMenu color='white' className='menu' onClick={() => {toggleSideBar()}} />
        <Sidebar showSideBar={ showSideBar } />
        <Routes>
          <Route path='/' exact element={<Illustrations />} />
          <Route path='/about' exact element={<About />} />
          <Route path='/contact' exact element={<Contact />} />
        </Routes>
      </BrowserRouter>
    </div>
  );
}

export default App;
