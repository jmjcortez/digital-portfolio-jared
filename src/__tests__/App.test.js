import '@testing-library/jest-dom';

import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

import App from '../App';

afterEach(() => {
  cleanup();
});

test('should render App', () => {
  render(<App />);
  const appElement = screen.getByTestId('App');
  expect(appElement).toBeInTheDocument();
});

test('matches snapshot', () => {
  const tree = renderer.create(<App />);
  expect(tree).toMatchSnapshot();
});