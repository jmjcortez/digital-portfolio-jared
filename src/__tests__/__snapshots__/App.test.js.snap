// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="App"
  data-testid="App"
>
  <svg
    className="menu"
    color="white"
    fill="currentColor"
    height="1em"
    onClick={[Function]}
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": "white",
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M4 6h16v2H4zm0 5h16v2H4zm0 5h16v2H4z"
    />
  </svg>
  <div>
    <div
      className="sidebar"
      data-testid="sidebar"
    >
      <span
        className="title"
      >
        Jared 
        <br />
         Cortez
      </span>
      <a
        className="links"
        href="/"
        onClick={[Function]}
      >
        <div
          className="link-text"
        >
          illustrations
        </div>
      </a>
      <a
        className="links"
        href="/contact"
        onClick={[Function]}
      >
        <div
          className="link-text"
        >
          contact
        </div>
      </a>
      <a
        className="links"
        href="/"
        onClick={[Function]}
      >
        <div
          className="link-text"
        >
          instagram
        </div>
      </a>
      <a
        className="links"
        href="/about"
        onClick={[Function]}
      >
        <div
          className="link-text"
        >
          about
        </div>
      </a>
      <div
        className="footer"
      >
        <span
          className="footer-text"
        >
          All images copyright Â©
        </span>
        <span
          className="footer-text"
        >
          Jared Cortez, 2022
        </span>
        <span
          className="footer-text"
        >
          All rights reserved
        </span>
      </div>
    </div>
  </div>
  <div
    className="gallery"
    data-testid="gallery"
  >
    <div
      className="piece-container"
    >
      <img
        alt="done_crying"
        className="piece"
        src="done_crying.png"
      />
      <div
        className="caption"
      >
        <h1>
          (done) crying
        </h1>
        <p>
          reflection of anxiety in someone
        </p>
      </div>
    </div>
    <div
      className="piece-container"
    >
      <img
        alt="3am_vibes"
        className="piece"
        src="3am_vibes.png"
      />
      <div
        className="caption"
      >
        <h1>
          (done) crying
        </h1>
        <p>
          reflection of anxiety in someone
        </p>
      </div>
    </div>
    <div
      className="piece-container"
    >
      <img
        alt="kamn"
        className="piece"
        src="kamn.png"
      />
      <div
        className="caption"
      >
        <h1>
          (done) crying
        </h1>
        <p>
          reflection of anxiety in someone
        </p>
      </div>
    </div>
    <div
      className="piece-container"
    >
      <img
        alt="colours"
        className="piece"
        src="colours.png"
      />
      <div
        className="caption"
      >
        <h1>
          (done) crying
        </h1>
        <p>
          reflection of anxiety in someone
        </p>
      </div>
    </div>
    <div
      className="piece-container"
    >
      <img
        alt="mental_torture"
        className="piece"
        src="mental_torture.png"
      />
      <div
        className="caption"
      >
        <h1>
          (done) crying
        </h1>
        <p>
          reflection of anxiety in someone
        </p>
      </div>
    </div>
    <div
      className="piece-container"
    >
      <img
        alt="hands_down"
        className="piece"
        src="hands_down.png"
      />
      <div
        className="caption"
      >
        <h1>
          (done) crying
        </h1>
        <p>
          reflection of anxiety in someone
        </p>
      </div>
    </div>
    <div
      className="piece-container"
    >
      <img
        alt="i_hope"
        className="piece"
        src="i_hope_to_be_around.png"
      />
      <div
        className="caption"
      >
        <h1>
          (done) crying
        </h1>
        <p>
          reflection of anxiety in someone
        </p>
      </div>
    </div>
    <div
      className="piece-container"
    >
      <img
        alt="griselda"
        className="piece"
        src="griselda.png"
      />
      <div
        className="caption"
      >
        <h1>
          (done) crying
        </h1>
        <p>
          reflection of anxiety in someone
        </p>
      </div>
    </div>
  </div>
</div>
`;
